sdk/containerregistry/Azure.Containers.ContainerRegistry/src/Helpers/ContainerRepository.cs

```diff
@@ -0,0 +1,271 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Core;
+using Azure.Core.Pipeline;
+
+namespace Azure.Containers.ContainerRegistry
+{
+    /// <summary> A helper class that groups information and operations about a repository in this container registry. </summary>
+    public partial class ContainerRepository
+    {
+        private readonly ClientDiagnostics _clientDiagnostics;
+        private readonly ContainerRegistryRestClient _restClient;
+
+        private readonly Uri _registryUri;
+        private readonly string _name;
+        private readonly string _fullyQualifiedName;
+
+        /// <summary>
+        /// Gets the Registry Uri.
+        /// </summary>
+        public virtual Uri RegistryUri => _registryUri;
+
+        /// <summary>
+        /// Gets the name of the repository.
+        /// </summary>
+        public virtual string Name => _name;
+
+        /// <summary>
+        /// Gets the fully qualified name of this repository.
+        /// </summary>
+        public virtual string FullyQualifiedName => _fullyQualifiedName;
+
+        /// <summary>
+        /// </summary>
+        internal ContainerRepository(Uri registryUri, string name,  ClientDiagnostics clientDiagnostics, ContainerRegistryRestClient restClient)
```

Comment:
This should have suitable public constructors as well - similar to this, but with ClientOptions. Could probably use the same one as the `ContainerRegistryClient`. That's what Storage, Search, and some others have done.

Reply:
See comment above.